spec/models/principal_spec.rb

```diff
@@ -202,4 +202,325 @@
       expect(Firm.where(id: firm.id)).to be_empty
     end
   end
+
+  describe '#next_onboarding_action' do
+    context 'when principal has no parent firm' do
+      before :each do
+        principal.firm.destroy
+        principal.reload
+      end
+
+      it 'returns :complete_a_firm' do
+        expect(principal.next_onboarding_action).to eql(:complete_a_firm)
+      end
+    end
+
+    context 'when principal only has a parent firm' do
+      before :each do
+        expect(principal.firm).not_to be_nil
+        expect(Firm.where(fca_number: principal.fca_number).count).to eql(1)
+      end
+
+      context 'and the parent firm has no email address' do
```

Comment:
This is confusing. The question we should be asking here is: do we consider the firm to be fully registered yet? Email address being nil is misleading and a bit leaky.

The fact that firms consider themselves registered if their `email_address` has been filled out is a shameful truth that firms would rather keep secret. They don't want us telling other bits of the system or relying on this knowledge. It's embarrassing for firms. Firms would rather you ask "are you `registered?`"

Seriously though, I feel it's time to encapsulate this logic in one place.